### PR TITLE
Add state to glossary

### PR DESCRIPTION
**Pull request by c0ffeeca7**
## Proposed change

Add term _state_ to glossary

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_data/glossary.yml [hunk 1]
```diff
@@ -275,6 +275,15 @@
     Services are called to perform actions.
   link: /docs/scripts/service-calls/
 
+- term: State
+  definition: |-
+    State is a data container holding the current state of an entity. E.g. if a light is on or off, the current temperature, or the RGB value of a light. The data type of state is `String`. States have an entity id, a state value, and 2 timestamps: `last_updated` and `last_changed`. Apart from the actual state value, a state can have attributes and context.
```
Comment (frenck) [inline source/_data/glossary.yml]: The "State is a data container holding" a bunch of things, v.s. "The data type of state is `String`" is conflicting in this sentence.

I think the main issue is the confusion with a full-state object in this case? From a state machine perspective, the state objects are the entities @ Runtime. I don't think for the end-user part, we need to maintain that difference.

From a user perspective, I think we can refer to the state object as "the entity".

The state of an entity is, indeed, always a string.

Comment (c0ffeeca7) [reply source/_data/glossary.yml]: Ah, yes. I did confuse it with the state object. Let me try this again.

File: source/_data/glossary.yml [hunk 2]
```diff
@@ -292,6 +292,14 @@
 
   link: /docs/scripts/service-calls/
 
+- term: State
+  definition: |-
+    State holds the information of interest of an entity. E.g. if a light is on or off, the current temperature, or the amount of energy used. The data type of state is `String`. Entities store 2 timestamps related to state: `last_updated` and `last_changed`. Each entity has exactly one state and state only holds one value at a time. However, entities can store attributes related to that state. E.g. the state of a light is _on_, and the related attributes are `brightness: 102`, `friendly name: Coffee light`.
```
Comment (frenck) [inline source/_data/glossary.yml]: - Small wording corrections.
- Added a small note/clarification on what a string is
- Removed the more concrete examples of state attributes in the end to a more human-readable text. This was mostly triggered by the `friendly_name` (as it isn't really an actual enlightenment on the state value).

```suggestion
    The state holds the information of interest of an entity. E.g. if a light is on or off, the current temperature, or the amount of energy used. The data type of state is `string` (a textual value). Entities store 2 timestamps related to the state: `last_updated` and `last_changed`. Each entity has exactly one state and the state only holds one value at a time. However, entities can store attributes related to that state. E.g. the state of a light is _on_, and the related state attributes could be its current brightness and color values.
```

Let me know what you think.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_data/glossary.yml (hunk 2 above)
Comment (c0ffeeca7) [reply source/_data/glossary.yml]: Love the improvements. :rocket: 
I think I need to remind myself from time to time that it is _user_ documentation we are doing here ;-)

**Comment by c0ffeeca7** [comment]
Added information that states are used by triggers and conditions.

File: source/_data/glossary.yml [hunk 3]
```diff
@@ -292,6 +292,16 @@
...
 
+- term: State
+  definition: |-
+    The state holds the information of interest of an entity. E.g. if a light is on or off, the current temperature, or the amount of energy used. The data type of state is `string` (a textual value). Entities store 2 timestamps related to the state: `last_updated` and `last_changed`. Each entity has exactly one state and the state only holds one value at a time. However, entities can store attributes related to that state. E.g. the state of a light is _on_, and the related state attributes could be its current brightness and color values.
+
+    State changes can be used as the source of triggers. The current state can be used in conditions.
+    
+  link: /docs/scripts/configuration/state_object/
+  excerpt: >
+    State holds the information of interest of an entity. E.g. if a light is on or off. Each entity has exactly one state and state only holds one value at a time. However, entities can store attributes related to that state. E.g. friendly name, unit of measurement, etc.
```
Comment (frenck) [inline source/_data/glossary.yml]: Small tweaks

```suggestion
    The state holds the information of interest of an entity. E.g. if a light is on or off. Each entity has exactly one state and the state only holds one value at a time. However, entities can store attributes related to that state. E.g. brightness, color, a unit of measurement, etc.
```

**Review by frenck — CHANGES_REQUESTED** [review]
Small tweak, otherwise looks good. 

You could consider avoiding `etc` and `e.g.`.

File: source/_data/glossary.yml (hunk 3 above)
Comment (c0ffeeca7) [reply source/_data/glossary.yml]: Thank you for the catch! Included these changes in the new commit to replace the _e.g._

File: source/_data/glossary.yml [hunk 4]
```diff
@@ -275,23 +275,33 @@
     The term service has 2 meanings in Home Assistant:
 
     **The information service**
-    E.g. the municipal waste management service that provides entities for organic, paper, and packaging waste. In terms of functionality, the information service is like a device. It is called *service* to avoid confusion, as it does not come with a piece of hardware.
+    For example, the municipal waste management service that provides entities for organic, paper, and packaging waste. In terms of functionality, the information service is like a device. It is called *service* to avoid confusion, as it does not come with a piece of hardware.
 
     **The software function that interacts with targets to make something happen**
-    A service carries out one specific task: e.g. turning on the light in the living room or sending a notification to a mobile phone. 
+    A service carries out one specific task: for example, turning on the light in the living room or sending a notification to a mobile phone.
```
Comment (frenck) [inline source/_data/glossary.yml]:
```suggestion
    A service carries out one specific task, for example: turning on the light in the living room or sending a notification to a mobile phone. 
```

File: source/_data/glossary.yml [hunk 5]
```diff
@@ -275,23 +275,33 @@
...
 
-    A service has targets and data. Service targets are: areas, devices, and entities. Service data carries the information required to define the desired state change in the target. E.g. the target, together with brightness 150 and RGB color `[255,0,0]`, or the message “Your coffee is ready”. 
+    A service has targets and data. Service targets are: areas, devices, and entities. Service data carries the information required to define the desired state change in the target. For example, the target, together with brightness 150 and RGB color `[255,0,0]`, or the message “Your coffee is ready”. 
 
     Services can be used in, for example, automation, scripts, dashboards, or voice commands to control your home.
 
     Home Assistant provides a series of predefined services, such as `homeassistant.turn_on`, `homeassistant.toggle`, or `homeassistant.reload`.
   excerpt: >
-    A service carries out one specific task: e.g. turn on the light in the 
+    A service carries out one specific task: For example, turn on the light in the
```
Comment (frenck) [inline source/_data/glossary.yml]:
```suggestion
    A service carries out one specific task, for example: turn on the light in the 
```

File: source/_data/glossary.yml [hunk 6]
```diff
@@ -275,23 +275,33 @@
...
 
+- term: State
+  definition: |-
+    The state holds the information of interest of an entity. For example, if a light is on or off, the current temperature, or the amount of energy used. The data type of state is `string` (a textual value). Entities store 2 timestamps related to the state: `last_updated` and `last_changed`. Each entity has exactly one state and the state only holds one value at a time. However, entities can store attributes related to that state. For example, the state of a light is _on_, and the related state attributes could be its current brightness and color values.
+
+    State changes can be used as the source of triggers. The current state can be used in conditions.
+    
+  link: /docs/scripts/configuration/state_object/
+  excerpt: >
+    The state holds the information of interest of an entity, for example if a light is on or off. Each entity has exactly one state and the state only holds one value at a time. However, entities can store attributes related to that state such as brightness, color, or a unit of measurement.
```
Comment (frenck) [inline source/_data/glossary.yml]:
```suggestion
    The state holds the information of interest of an entity, for example, if a light is on or off. Each entity has exactly one state and the state only holds one value at a time. However, entities can store attributes related to that state such as brightness, color, or a unit of measurement.
```

File: source/_data/glossary.yml [hunk 7]
```diff
@@ -319,7 +329,7 @@
 - term: Variables
   definition: >-
     Variables are used to store values in memory that can be processed
-    e.g. in a script.
+    for example in a script.
```
Comment (frenck) [inline source/_data/glossary.yml]:
```suggestion
    for example, in a script.
```

**Review by frenck — CHANGES_REQUESTED** [review]

**Review by frenck — APPROVED** [review]
LGTM ✅

**Pull request merged**